frontend/src/components/common/input_password/password.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { Input } from 'components/common/common';
+import {
+  FormControl,
+  FormControlErrors,
+  MasterSignUpDto,
+} from 'common/types/types';
+import { InputType } from 'common/enums/enums';
+import { getNameOf } from 'helpers/helpers';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEyeSlash, faEye } from '@fortawesome/free-solid-svg-icons';
+
+type Props = {
+  control: FormControl;
+  errors: FormControlErrors;
+};
+
+const Password: React.FC<Props> = ({ control, errors }) => {
+  const [show, setShow] = useState(false);
+
+  return (
+    <div>
+      <Input
+        type={show ? InputType.TEXT : InputType.PASSWORD}
+        label={'Password'}
+        placeholder={'Enter your password'}
+        name={getNameOf<MasterSignUpDto>('password')}
+        control={control}
+        errors={errors}
+      />
+      <button
+        onClick={(): void => {
+          setShow(!show);
+        }}
```

Comment:
It's better to avoid anonymous functions. You can create handle function like toggleShow() and use it

Reply:
`handleShowToggle` 😉